src/syrupy/extensions/json/__init__.py

```diff
@@ -133,6 +137,12 @@ def _filter(
         if isinstance(data, (datetime.datetime,)):
             return data.strftime("%Y-%m-%dT%H:%M:%S.%f%z")
 
+        if isinstance(data, FunctionType):
+            return (
+                f"<{FunctionType.__name__} "
+                f"'{data.__qualname__}{str(inspect.signature(data))}'>"
```

Comment:
Serialization nitpick
```suggestion
                f"{data.__qualname__}{str(inspect.signature(data))}>"
```
Result
```json
"<function function_to_test(var1, var2='test_val', var3: str = 'test_val2', *, kwvar1, kwvar2='some_val') -> str>"
```